### PR TITLE
CLOUDP-196443 Update mongot image to fetch the correct link

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes
Update mongot image to fetch the correct link

_Jira ticket:_ CLOUDP-196443

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/deployments/options/deployment_opts.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ const (
 	spinnerSpeed         = 100 * time.Millisecond
 	// based on https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Clusters/operation/createCluster
 	clusterNamePattern    = "^[a-zA-Z0-9][a-zA-Z0-9-]*$"
-	MongotDockerImageName = "docker.io/mongodb/apix_test:mongot-preview"
+	MongotDockerImageName = "docker.io/mongodb/mongodb-atlas-search:latest"
```
Comment (andreaangiolillo) [inline internal/cli/atlas/deployments/options/deployment_opts.go:36]: is it safe to get always the latest version? should we ping to a version to not get breaking changes?

**Review by andreaangiolillo — DISMISSED** [review]
LGTM

File: internal/cli/atlas/deployments/scripts/start_mongot.sh [hunk 2]
```diff
@@ -14,11 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-if [ ! -f "$KEYFILEPATH" ]
-then
-    echo "$KEYFILECONTENTS" > "$KEYFILEPATH"
-    chmod 400 "$KEYFILEPATH"
```
Comment (tibulca) [inline internal/cli/atlas/deployments/scripts/start_mongot.sh:20]: is mongot not checking anymore the file permissions? (I think it was failing if permissions are too wide)

File: internal/cli/atlas/deployments/options/deployment_opts.go (hunk 1 above)
Comment (fmenezes) [reply internal/cli/atlas/deployments/options/deployment_opts.go:36]: for now we only have one version `preview`, once atlas search takes over we might know more

File: internal/cli/atlas/deployments/scripts/start_mongot.sh (hunk 2 above)
Comment (fmenezes) [reply internal/cli/atlas/deployments/scripts/start_mongot.sh:20]: it is done inside container

File: internal/cli/atlas/deployments/scripts/start_mongot.sh [hunk 3]
```diff
@@ -14,11 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-if [ ! -f "$KEYFILEPATH" ]
```
Comment (tibulca) [inline internal/cli/atlas/deployments/scripts/start_mongot.sh:17]: without this check, the restarting a container was failing, is this not the case anymore?

File: internal/cli/atlas/deployments/scripts/start_mongot.sh (hunk 2 above)
Comment (fmenezes) [reply internal/cli/atlas/deployments/scripts/start_mongot.sh:20]: https://github.com/10gen/mongot/pull/1802/files#diff-997d64eedc645601c81b86be6b25f569abfff63aaa3a49ff88975988a14065fbR177

File: internal/cli/atlas/deployments/scripts/start_mongot.sh (hunk 3 above)
Comment (fmenezes) [reply internal/cli/atlas/deployments/scripts/start_mongot.sh:17]: not the case, I've tested it

**Review by tibulca — APPROVED** [review]
LGTM if [atlas_deployments_e2e](https://spruce.mongodb.com/task/mongodb_atlas_cli_master_e2e_atlas_deployments_atlas_deployments_e2e_patch_ec2f19bf8b9a2f53e6f7eb64beab0dae8f9f0373_650472e00305b93e52caee12_23_09_15_15_06_10/logs?execution=0) is succesfull

File: internal/cli/atlas/deployments/scripts/start_mongot.sh (hunk 3 above)
Comment (fmenezes) [reply internal/cli/atlas/deployments/scripts/start_mongot.sh:17]: also @tibulca I think we should make a test for this, it seems an important enough regression for us to test

**Pull request merged**